### DISCUSSION TITLE
AUTO v SEQUENCE when switching between sqlite to postgres

### DISCUSSION THREAD

**Discussion by tacman** [symfony/symfony#59038] [Q&A] [answered]
I often use SQLite for my database when prototyping a project, and then use Postgres when deploying.  I like being able to switch, especially if I have outside developers (like designers) who are working on something unrelated to the database and I don't want them to need to install docker or postgres.

It used to be that I could simply define an "AUTO" strategy.

```php
    #[ORM\Id]
    #[ORM\GeneratedValue('AUTO')]
    #[ORM\Column(type: 'integer')]
    private int $id;
```

This works  with sqlite, but not postgres, which wants "SEQUENCE", which doesn't work with sqlite.

I feel like I must be missing something, how can I work with sqlite OR postgres during development?

**Comment by stof** [comment, marked as answer]
In doctrine/orm 2.17+, you can configure your preference for the way `AUTO` strategy gets implemented (and it is deprecated not to configure it for platforms that support multiple ones like postgresql because defaults are changing in 3.0).
If you configure those identity generation preferences (which requires using DoctrineBundle 2.13+ to have easy access to this configuration through the bundle config), you can still use `AUTO` as strategy in your mapping.

**Comment by tacman** [comment]
Thanks!  Indeed, it was an older project that had been migrated from doctrine 2.
